### DISCUSSION TITLE
Virtual chassie and module types confusion

### DISCUSSION THREAD

**Discussion by chuegel** [netbox-community/netbox#12672] [Help Wanted!]
Hey there,

following situation:

we have a couple of Cisco 6509-E switches with the same interface line cards (WS-X6148A-GE-45AF) and same SUPs configured.
The way we provisioned them into netbox is as follows:
```

1. Create device type
- manufacturer: Cisco
- model: 6509-E
- height: 14
- part_number: WS-6509-E
- subdevice_role: parent
- is_full_depth: true
- airflow: left-to-right

2. Create module bay
- device: 6509-E
- name: Slot-[1-9]
- position: [1-9]

3. Create module type for interface line card
- manufacturer: Cisco
- model: WS-X6148A-GE-45AF
- part_number: WS-X6148A-GE-45AF

4. Create module type for SUP
- manufacturer: Cisco
- model: WS-SUP32-10GE-3B
- part_number: WS-SUP32-10GE-3B

5. Create interface template for line card WS-X6148A-GE-45AF
- device_type: ---
- module: WS-X6148A-GE-45AF
- name: Gi{module}/[1-48]
- type: 1000BASE-T
- poe_mode: PD
- poe_type: 802.3af

5. Create 10G interface template for line card WS-X6148A-GE-45AF
- device_type: ---
- module: WS-SUP32-10GE-3B
- name: Te{module}/[1-2]
- type: 10GBASE-T

6. Create 1G interface template for line card WS-X6148A-GE-45AF
- device_type: ---
- module: WS-SUP32-10GE-3B
- name: Ge{module}/[3]
- type: 1000BASE-T

7. Add 6509-E device
- name: Cat6k-Test
- device_role: Core Switch
- manufacturer: Cisco
- device_type: 6509-E

8. Add modules to module bay
- device: Cat6k-Test
- module_bay: Slot-1
- manufacturer: Cisco
- module_type: WS-X6148A-GE-45AF

9. Finally add SUP module to module bay
- device: Cat6k-Test
- module_bay: Slot-5
- manufacturer: Cisco
- module_type: WS-SUP32-10GE-3B
```

Everything works as expected: the interfaces are correctly populated according to the module type in the module bay slot.

But.... ;)

It happens that we have a pair of the same devices and same linecards in a VSS. When I create the devices (see above) and add them to a virtual chassie the interfaces obviously do not match anymore (become duplicates) i.e:

Gi{module}/1 is Gi1/1 on the first member
Gi{module}/1 is Gi1/1 on the second member

On the VSS master I have two identical interfaces

The interfaces on the master VC should be:
{VC_member}/{module}/1 

Question: 
Is there a placeholder like {module} also for the VC member?
How can this be solved? I can of course create a module type for each member in the VSS like:

```
- manufacturer: Cisco
- model: WS-X6148A-GE-45AF-VSS-1
- part_number: WS-X6148A-GE-45AF

# add interfaces to module type
- device_type: ---
- module: WS-X6148A-GE-45AF-VSS-1
- name: Gi1/{module}/[1-48]
- type: 1000Base-T
- poe_mode: PD
- poe_type: 802.3af

- manufacturer: Cisco
- model: WS-X6148A-GE-45AF-VSS-2
- part_number: WS-X6148A-GE-45AF

# add interfaces to module type
- device_type: ---
- module: WS-X6148A-GE-45AF-VSS-2
- name: Gi2/{module}/[1-48]
- type: 1000Base-T
- poe_mode: PD
- poe_type: 802.3af

```
etc. but this seems to be a ugly way to do this.

I hope I made myself clear. Any hints are greatly appreciated.

Cheers

**Comment by candlerb** [comment]
No placeholder at the moment. You need to manually rename the interfaces before joining the devices into a virtual chassis: there is a "bulk rename" feature for this (e.g. Te1/x/x to Te2/x/x)

**Reply by chuegel** [reply]
Ok, thanks!

**Reply by ctsalidis** [reply]
Hi @candlerb @chuegel, do you know whether there is any update in regards to the VC_member placeholder? Do you still create the devices and cards in this format (e.g. Te1/x/x to Te2/x/x) and then join them into a virtual chassis? Thank you!

**Reply by chuegel** [reply]
Still renaming the interfaces before joining

**Reply by ctsalidis** [reply]
Thank you @chuegel! Perhaps you could help me with another question I have. Is it possible to get all the interfaces of both nodes using the virtual chassis name?

**Reply by chuegel** [reply]
Yes, you can: Device Components > Interfaces or via a API query

**Reply by ctsalidis** [reply]
Thank you @chuegel !

**Comment by z15858167** [comment]
When will this problem be solved, wow, now many core devices are stacked with 2

**Reply by candlerb** [reply]
If you are doing this repeatedly, and you find it's too inconvenient for you to rename the interfaces of the second device before joining the second device into a stack with the first, then you can write a Custom Script. In one operation it can:
* create the two devices
* insert the modules
* rename the interfaces
* create the VSS
